node-manager/src/node.rs

```diff
@@ -522,8 +522,20 @@ impl Node {
     }
 
     /// pay_invoice sends off the payment but does not wait for results
-    pub fn pay_invoice(&self, invoice: Invoice) -> Result<MutinyInvoice, MutinyError> {
-        let pay_result = self.invoice_payer.pay_invoice(&invoice);
+    pub fn pay_invoice(
+        &self,
+        invoice: Invoice,
+        amt_sats: Option<u64>,
+    ) -> Result<MutinyInvoice, MutinyError> {
+        let pay_result = if invoice.amount_milli_satoshis().is_none() {
+            if amt_sats.is_none() {
+                return Err(MutinyError::InvoiceInvalid);
+            }
+            self.invoice_payer
+                .pay_zero_value_invoice(&invoice, amt_sats.unwrap())
+        } else {
+            self.invoice_payer.pay_invoice(&invoice)
```

Comment:
we should probably throw an error if `amt_sats`is some here because the frontend is probably misusing the api

Reply:
added `is_some()`